pyroomacoustics/open_sofa_interpolate.py

```diff
@@ -0,0 +1,556 @@
+
+
+
+                no_of_lat = list(collections.Counter(theta_rcv).values())[0]
+
+                #no_of_long = len(no_of_lat)
+
+                theta_16 = np.array([theta_rcv[i] for i in range(len(theta_rcv)) if i % no_of_lat == 0])
+
+
+                return phi_rcv, theta_rcv, r_rcv, theta_16, rcv_msr, no_of_nodes, samples, no_of_lat
+            else:
+                file_sofa = sofa.Database.open(path)
+                # Source positions
+
+                src_pos = file_sofa.Source.Position.get_values()
+
+                # CHEAP HACCCKK SPECIFICALLY FOR DIRPAT
+
+                src_pos_RS = np.reshape(src_pos, [30, 16, 3])
+
+                src_pos = np.swapaxes(src_pos_RS, 0, 1).reshape([480, -1])
+
+                ###########################
+
+                # Get impulse responses from all the measurements
+
+                IR_S = file_sofa.Data.IR.get_values()
+
+                # Look for receiver of specific type requested by user"
+
+                rcv_msr = IR_S[:, self.id,:]  # First receiver #Shape (no_measurement_points * no_receivers * no_samples_IR)
+
+                if "Debug_sofa_file" not in self.path:
+                    rcv_msr = decimate(rcv_msr, int(round(file_sofa.Data.SamplingRate.get_values()[0] / fs)), axis=-1)
+                    print("Decimating FIR'S Receiver")
+
+
+                no_of_nodes = file_sofa.Dimensions.M  # Number of measurement points
+
+                # samples = file_sofa.Dimensions.N  # Samples per IR
+                samples = rcv_msr.shape[1]
+
+                # All measurements should be in = radians phi [0,2*np.pi] , theta [0,np.pi]
+
+                phi_src = src_pos[:, 0]
+                theta_src = src_pos[:, 1]
+                r_src = src_pos[:, 2]
+
+                no_of_lat = list(collections.Counter(theta_src).values())[0]
+
+                #no_of_long = len(no_of_lat)
+
+                theta_16 = np.array([theta_src[i] for i in range(len(theta_src)) if i % no_of_lat == 0])
+
+                return phi_src, theta_src, r_src, theta_16, rcv_msr, no_of_nodes, samples, no_of_lat
+
+    def cal_index_knn(self,azimuth,colatitude):
+
+        """
+        Calculates nearest neighbour for all the image source for the source and the receiver
+        This method exist so that we don't have to query each image source , nearest neighbour for all the incoming angles (mic) and departure angles(sources)
+        are calculated at once.
+        : Parameter
+        ------------------------------
+
+        azimuth:nd.array() (int)  [-pi,pi]
+        colatitude:nd.array() (int) [0,pi]
+
+        : Return : nd.array() return nearest index on the grid based on the input angles using kDtree
+        """
+
+        longitude=azimuth.reshape(-1,1)
+        latitude=colatitude.reshape(-1,1)
+
+        start=timer()
+        if self.interpolate:
+            dd,ii=self.nn_kd_tree_rotated_fibo_grid.query(np.hstack((longitude,latitude)))
```

Comment:
So my proposal of apply rotation to query point would be to rotate longitude/latitude here. This would be easier to do if the grid is stored in cartesian coordinates.

Also, just a detail, but why add longitude/latitude notation here ? Maybe should stick to azimuth/coltitude throughout.